gapis/perfetto/android/trace.go

```diff
@@ -167,15 +175,33 @@ func Start(ctx context.Context, d adb.Device, a *android.ActivityAction, opts *s
 		}
 	}
 
+	// With the comsumer protocol, in Android 10 it causes a stall when
+	// traced_pros writes into file or the trace is too long. To be more
+	// conservative we consider traces longer than 10 seconds long traces.
+	// In this case, don't create perfetto client, instead fall back to CLI.
+	var c *perfetto.Client
+	if !opts.PerfettoConfig.GetWriteIntoFile() && opts.PerfettoConfig.GetDurationMs() <= maxInMemDuration {
```

Comment:
We don't need the duration check. If write into file is not set there is no benefit of using the command line even for long traces. The command line does exactly the same as we do and downloads the data at the end if write-into-file is disabled.